gwsurrogate/new/precessing_surrogate.py

```diff
@@ -764,21 +751,61 @@ def __call__(self, q, chiA, chiB, ellMax=4):
-            # m=0 is different
-            re = _eval_comp(self.data['%s_0_real'%(ell)], q, chiA, chiB)
-            im = _eval_comp(self.data['%s_0_imag'%(ell)], q, chiA, chiB)
-            modes[ell*(ell+1) - 4] = re + 1.j*im
-
+            
+            # NOTE: modes is populated with zeros, so skipping means 
+            #       "set to zero". This is required for inertial_waveform_modes,
+            #       which maps from coorb to inertial frame modes. 
+            # m=0 has a different evaluation pattern
+            if (ell,0) in self.mode_list:
+                re = self._eval_comp(self.data['%s_0_real'%(ell)], q, chiA, chiB)
+                im = self._eval_comp(self.data['%s_0_imag'%(ell)], q, chiA, chiB)
+                modes[ell*(ell+1) - 4] = re + 1.j*im
+                #print("evaluation ell=%s, m=0"%ell)
+
+            # NOTE: modes is populated with zeros, so skipping means 
+            # "set to zero". This is required for inertial_waveform_modes,
+            # which maps from coorb to inertial frame modes. 
             for m in range(1, ell+1):
-                rep = _eval_comp(self.data['%s_%s_Re+'%(ell, m)], q, chiA,chiB)
-                rem = _eval_comp(self.data['%s_%s_Re-'%(ell, m)], q, chiA,chiB)
-                imp = _eval_comp(self.data['%s_%s_Im+'%(ell, m)], q, chiA,chiB)
-                imm = _eval_comp(self.data['%s_%s_Im-'%(ell, m)], q, chiA,chiB)
-                h_posm, h_negm = _assemble_mode_pair(rep, rem, imp, imm)
-                modes[ell*(ell+1) - 4 + m] = h_posm
-                modes[ell*(ell+1) - 4 - m] = h_negm
+                if (ell,m) in self.mode_list:                
+                    rep = self._eval_comp(self.data['%s_%s_Re+'%(ell, m)], q, chiA,chiB)
+                    rem = self._eval_comp(self.data['%s_%s_Re-'%(ell, m)], q, chiA,chiB)
+                    imp = self._eval_comp(self.data['%s_%s_Im+'%(ell, m)], q, chiA,chiB)
+                    imm = self._eval_comp(self.data['%s_%s_Im-'%(ell, m)], q, chiA,chiB)
+                    h_posm, h_negm = _assemble_mode_pair(rep, rem, imp, imm)
+                    modes[ell*(ell+1) - 4 + m] = h_posm
+                    modes[ell*(ell+1) - 4 - m] = h_negm
+                    #print("evaluation ell=%s, m=%s"%(ell,m))
 
         return modes
+    
+    def _eval_comp(self, data, q, chiA, chiB):
+        nodes = []
+        for orders, coefs, ni in zip(data['orders'], data['coefs'],
+                data['nodeIndices']):
+
+            fit_data = {
+                'bfOrders': orders,
+                'coefs': coefs,
+                }
+            x = np.append(q, np.append(chiA[ni], chiB[ni]))
+            fit_params = self._get_fit_params(x)
+            nodes.append(_eval_scalar_fit(fit_data, fit_params, self._get_fit_settings))
+
+        return np.array(nodes).dot(data['EI_basis'])
+    
+    def _check_h5group_exists(self, h5file, group_name):
+        """ Check if h5 group GROUP_NAME has valid data to load. """
+
```

Comment:
Can this be as simple as:
```
return group_name in h5file
```
Based on https://forum.hdfgroup.org/t/how-to-check-if-a-group-path-is-valid/8620